include/sys/arc.h

```diff
@@ -110,20 +121,27 @@ typedef enum arc_flags
 	ARC_FLAG_L2_WRITING		= 1 << 11,	/* write in progress */
 	ARC_FLAG_L2_EVICTED		= 1 << 12,	/* evicted during I/O */
 	ARC_FLAG_L2_WRITE_HEAD		= 1 << 13,	/* head of write list */
+	/*
+	 * Encrypted or authenticated on disk (may be plaintext in memory).
+	 * This header has b_crypt_hdr allocated.
+	 */
+	ARC_FLAG_PROTECTED		= 1 << 14,
```

Comment:
oh, I see.  Maybe include a note in the comment `(does not include indirect blocks even if they have a checksum of MACs and thus have the X (encrypted) bit set in their BP)`